Description: fix memory leak with threaded renderer
 This patch is actually the merge of https://codereview.qt-project.org/#/c/200715/
 and https://codereview.qt-project.org/#/c/202781/ and will be included in the
 next minor release.
Bug-Qt: https://bugreports.qt.io/browse/QTBUG-61754
Author: Mitch Curtis

---
 src/quick/scenegraph/qsgthreadedrenderloop.cpp |    1 +
 1 file changed, 1 insertion(+)

--- a/src/quick/scenegraph/qsgthreadedrenderloop.cpp
+++ b/src/quick/scenegraph/qsgthreadedrenderloop.cpp
@@ -553,6 +553,7 @@ void QSGRenderThread::sync(bool inExpose
         if (d->renderer)
             d->renderer->clearChangedFlag();
         d->syncSceneGraph();
+        sgrc->endSync();
         if (!hadRenderer && d->renderer) {
             qCDebug(QSG_LOG_RENDERLOOP) << QSG_RT_PAD << "- renderer was created";
             syncResultedInChanges = true;
